### PR TITLE
[GB upgrade e2e] Fill all the popular blocks with the necessary minimum data for them to appear in the frontend and make sure each of them is displayed

### PR DESCRIPTION
**Pull request by fullofcaffeine**
### What 

Fill all the popular blocks with the necessary minimum data for them to appear in the frontend and make sure each of them is displayed.

### Why

The[ initial task](https://github.com/Automattic/wp-calypso/pull/45048) added the framework/PoC for automating our manual testing process. It only configured two blocks, though. The goal of this PR is to setup all of the popular blocks being tested as part of the upgrade E2E test, and verify them in the published page too (edge/non-edge and across the most popular themes)

Tracking issue: https://github.com/Automattic/wp-calypso/issues/45209 (_Fill all the popular blocks with the necessary minimum data for them to appear in the frontend and make sure each of them is displayed_)

### TODO
* [ ] Fix issues reported in CI
* [ ] Squash commits _(due to the nature of this changeset, where I need to test on CI, there are more commits than I'd like it to have, so some hygiene before merging would be nice)_

### CODE REVIEW

File: test/e2e/lib/gutenberg/blocks/layout-grid-block-component.js [hunk 1]
```diff
@@ -1,11 +1,72 @@
...
+	 * I'm not sure if this belongs here or extracted into a method in the GutenbergEditorComponent or maybe
+	 * its own QuickInserter class. I'll figure out later.
+	 * Also, the block wrapper div changes after you setup the number of columns you want. This makes it tricky
+	 * to scope the elements to the original block. I'm for now brute-forcing the selection by using a more
+	 * general selector.
+	 *
+	 * This probably won't support multiple instances of the same block. It hasn't been tested
+	 * with multiple layout grids on the page either, but it works fine for simple scenarios.
+	 *
+	 * @param { typeof GutenbergBlockComponent } blockClass A block class that responds to blockTitle and blockName
```
Comment (fullofcaffeine) [inline test/e2e/lib/gutenberg/blocks/layout-grid-block-component.js:38]: See: https://github.com/jsdoc/jsdoc/issues/1349

**Comment by fullofcaffeine** [comment]
@Automattic/good-mountain This is almost ready. I still want to spend a bit more time testing in order to make sure the test is stable and not flaky, but it seems to work fine. 

All the blocks (that we test manually) that require some configuration in the editor are being set up so that they show some meaningful data in the frontend. The test also verifies for their presence in the frontend. I'm not going into deep frontend testing yet, as this will be done as part of the next task (see the tracking issue).

This goes a step further into automating our manual checklist, and will already be useful for testing the next release of GB on dotcom.

File: test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
  */
 import assert from 'assert';
 import config from 'config';
-import { times } from 'lodash';
+import { times, sample } from 'lodash';
```
Comment (fullofcaffeine) [inline test/e2e/specs/wp-calypso-gutenberg-upgrade-spec.js:6]: This was added by vscode, wasn't intentional, I'll remove.

**Comment by fullofcaffeine** [comment]
Now that the tests [are running](https://github.com/Automattic/wp-calypso/pull/45284/commits/5f53475d64713ad715a63a70390d0a810180f726) on CI, I found out a couple of new issues with it: 

1) the screenshots, which work fine locally, do not work well on CI. I think that has to do with the relative path that the [writeScreenshot](https://github.com/Automattic/wp-calypso/blob/master/test/e2e/lib/media-helper.js#L77) function uses and the fact that magellan runs the test from a[ temp directory](https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/38353/workflows/3b8123ce-0646-4c95-9033-6cd801279645/jobs/101881) (?). 

2) It's now running across different screen sizes, and there are [some issues](https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/38355/workflows/04ed9684-5c16-4691-bc46-483637096971/jobs/101885) with the mobile viewport.

3) Some errors in the[ desktop viewport](https://app.circleci.com/pipelines/github/Automattic/wp-e2e-tests-for-branches/38354/workflows/30e8745a-c77b-4d6f-8010-f9389328a8ec/jobs/101884) that only happen on CI that I need to investigate and fix, too.

I'll keep working on fixing those tomorrow.

File: test/e2e/lib/gutenberg/blocks/dynamic-separator-block-component.js [hunk 3]
```diff
@@ -6,6 +11,48 @@ import GutenbergBlockComponent from './gutenberg-block-component';
...
 	static blockTitle = 'Dynamic HR';
 	static blockName = 'coblocks/dynamic-separator';
+	static blockFrontendSelector = By.css( '.entry-content .wp-block-coblocks-dynamic-separator' );
+
+	/**
+	 * Expands the horizontal ruler by the amount of pixels specified. This is useful
+	 * to test the functionality of the Dynamic HR in the editor and later in the
+	 * frontend too.
+	 *
+	 * @param {number} pixels the amount of pixerls to expand the HR ruler by, horizontally.
```
Comment (ockham) [inline test/e2e/lib/gutenberg/blocks/dynamic-separator-block-component.js:21]:
```suggestion
	 * @param {number} pixels the amount of pixels to expand the HR ruler by, horizontally.
```

**Comment by ockham** [comment]
: * [ ] Squash commits _(due to the nature of this changeset, where I need to test on CI, there are more commits than I'd like it to have, so some hygiene before merging would be nice)_

Don't bother about that; in most of our repos, we've limited the merge button to 'Squash and merge' anyway, so all commits in a PR will be squashed into one. That means that there's no point in polishing individual commits in a PR since that commit history won't be preserved when merging into `master` anyway.

**Comment by ockham** [comment]
This PR is kinda big :sweat_smile: Can we maybe split it up a bit? E.g. can we move the changes to the logic that takes the screenshots to a different PR? They're not directly related to filling the blocks with data, are they? :thinking: 

I've read your comments on the screenshots issue but I'm not sure I understand them :thinking: Screenshots were working prior to this PR, weren't they? So we shouldn't really need to touch them, should we? :thinking:

**Comment by ockham** [comment]
The `jetpack/*` blocks related test changes don't seem to be strictly upgrade related, so how about (additionally?) contributing them to Jetpack's own e2e tests so issues can be caught there? (They're running them against the latest GB plugin release AFAIK -- @brbrr knows more.)

**Comment by fullofcaffeine** [comment]
Hey @ockham, thanks for taking the time to review this PR! 🙇🏻 

*> Edit -- e.g. what happens in CI if you revert all the screenshot related changes?
Screenshots will not work on CI, test will fail because the directory is not found. More on that below.

> I've read your comments on the screenshots issue but I'm not sure I understand them thinking Screenshots were working prior to this PR, weren't they? So we shouldn't really need to touch them, should we? 

They were, but locally. I only figure out recently that I needed to apply the `@parallel` tag to the test in order for the CI to pick it up, and then I found out about the issue. I've documented it in my rubber-duck monologue, here: p1600379866056600-slack-C1A1EKDGQ. Once I got it to work, I found out that the screenshots were not available under the "Artifacts" tab in CircleCI, but I already found out how to solve that, more details here: p1600398576006500-slack-C1A1EKDGQ.

> This PR is kinda big sweat_smile Can we maybe split it up a bit? E.g. can we move the changes to the logic that takes the screenshots to a different PR? They're not directly related to filling the blocks with data, are they? thinking

The logic to take screenshots were added in a previous PR. The changes here are only for the aforementioned fixes. 

Most of the code here is not related to the screenshots, but they are related to the task of filling in the blocks (with sample data/assets) in the editor. We could perhaps split that into multiple PRs, per block? 

> Don't bother about that; in most of our repos, we've limited the merge button to 'Squash and merge' anyway, so all commits in a PR will be squashed into one. That means that there's no point in polishing individual commits in a PR since that commit history won't be preserved when merging into master anyway.

Sounds good. I didn't want the description of the merge commit to include so many commits, though, but I can live with that :)

**So, in sum, what's left?**

* Copy the screenshos from the magellan temp directory to the directory that is accessible in the CircleCI artifacts tab (p1600436756010100-slack-C1A1EKDGQ?thread_ts=1600398576.006500&cid=C1A1EKDGQ). We could leave that for another PR.
* Deal with the flakiness with the publish buttons. Sometimes it fails silently and I haven't been able to identify the root issue yet. More details here: p1600398296000300-slack-C017FDWM21W (in the "Anything blocking your progress section"). This needs to be done as part of this PR.

File: test/e2e/lib/mocha-hooks.js [hunk 4]
```diff
@@ -64,6 +64,7 @@ afterEach( async function () {
 	let filenameCallback;
 
 	if ( this.currentTest.state === 'failed' ) {
+		console.error( this.currentTest.err );
```
Comment (fullofcaffeine) [inline test/e2e/lib/mocha-hooks.js:67]: TODO Remove this before merging

**Comment by fullofcaffeine** [comment]
Closed in favor of the following PRs:

* https://github.com/Automattic/wp-calypso/pull/45794
* https://github.com/Automattic/wp-calypso/pull/45795
* https://github.com/Automattic/wp-calypso/pull/45796
* https://github.com/Automattic/wp-calypso/pull/45797
*  https://github.com/Automattic/wp-calypso/pull/45798
* https://github.com/Automattic/wp-calypso/pull/45799

**Pull request closed without merging**